src/services/transaction/TransactionDryRunService.ts

```diff
@@ -14,43 +14,47 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import { BlockHash } from '@polkadot/types/interfaces';
+import type { BlockHash, CallDryRunEffects, XcmDryRunApiError } from '@polkadot/types/interfaces';
+import type { Result } from '@polkadot/types-codec';
 
-import { ITransactionDryRun, TransactionResultType, ValidityErrorType } from '../../types/responses';
+import { ITransactionDryRun } from '../../types/responses';
 import { AbstractService } from '../AbstractService';
 import { extractCauseAndStack } from './extractCauseAndStack';
 
+export type SignedOriginCaller = {
+	System: {
+		Signed: string;
+	};
+};
+
 export class TransactionDryRunService extends AbstractService {
-	async dryRuntExtrinsic(hash: BlockHash, transaction: string): Promise<ITransactionDryRun> {
+	async dryRuntExtrinsic(
+		hash: BlockHash,
+		senderAddress: string,
+		transaction: `0x${string}`,
+	): Promise<ITransactionDryRun> {
 		const { api } = this;
 
 		try {
-			const [applyExtrinsicResult, { number }] = await Promise.all([
-				api.rpc.system.dryRun(transaction, hash),
+			const originCaller: SignedOriginCaller = {
+				System: {
+					Signed: senderAddress,
+				},
+			};
+
+			const [dryRunResponse, header] = await Promise.all([
```

Comment:
```suggestion
			const [dryRunResponse, { number }] = await Promise.all([
```
Small nit but can't this remain the same as before?

Reply:
you are right, updated!